src/Common/src/Microsoft/Win32/SafeHandles/SafeHeapHandle.cs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Win32.SafeHandles
+{
+    using System;
+    using System.Diagnostics;
+    using System.Runtime.InteropServices;
+
+    /// <summary>
+    /// Handle for heap memory
+    /// </summary>
+    /// <remarks>
+    /// Uses new Heap* methods instead of Local* methods, which are deprecated.  While both calls utilize the same underlying
+    /// heap allocation, Local* adds some overhead (*significant* overhead if LMEM_MOVEABLE is used). .NET forces LMEM_FIXED
+    /// with LocalAlloc for Marshal.AllocHGlobal so it doesn't hit the super slow path.
+    /// 
+    /// Windows attempts to grab space from the low fragmentation heap if the requested memory is below a platform specific
+    /// threshold and certain flags aren't in play (such as NO_SERIALIZE).
+    /// </remarks>
+    internal sealed class SafeHeapHandle : SafeBuffer
+    {
+        /// <summary>
+        /// The handle for the process heap.
+        /// </summary>
+        private static readonly IntPtr s_processHeap = Interop.mincore.GetProcessHeap();
```

Comment:
I suggest then that we use AllocHGlobal and friends rather than these custom P/Invokes.  And then do whatever we need to do to make AllocHGlobal perform better, though I've not seen measurements that suggest they're performing poorly or measurably worse than these custom P/Invokes.  If you have such data, please share.